data-serving/scripts/aggregate-data/aggregate/app.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import logging
+import os
+import sys
+import urllib
+from typing import Any, Callable, Optional
+
+import boto3
+import pandas as pd
+import pymongo
+import iso3166
+
+
+def upload(S3, data: str, bucket: str, keys: list[str]):
+    """Upload data to S3
+
+    data -- Data to upload as a string
+    bucket -- S3 bucket to upload to
+    keys -- List of S3 paths to upload to
+    """
+    try:
+        logging.info(f"Uploading data to {keys} in {bucket}")
+        for key in keys:
+            S3.put_object(
+                ACL="public-read",
+                Body=data,
+                Bucket=bucket,
+                Key=key,
+            )
+    except Exception as e:
+        logging.error(e)
+        logging.error(f"Failed to upload data to s3://{bucket}/{key}")
```

Comment:
`for key in keys:
  try: ...
  except ...
`

Also, you can condense those two `logging.error` calls with one using `logging.exception` https://docs.python.org/3/library/logging.html#logging.Logger.exception